src/neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -13,25 +13,48 @@ public class TransactionVerificationContext
         /// </summary>
         private readonly Dictionary<UInt160, BigInteger> senderFee = new Dictionary<UInt160, BigInteger>();
 
+        /// <summary>
+        /// Store oracle requests
+        /// </summary>
+        private readonly Dictionary<ulong, UInt256> oracleRequests = new Dictionary<ulong, UInt256>();
+
         public void AddTransaction(Transaction tx)
         {
             if (senderFee.TryGetValue(tx.Sender, out var value))
                 senderFee[tx.Sender] = value + tx.SystemFee + tx.NetworkFee;
             else
                 senderFee.Add(tx.Sender, tx.SystemFee + tx.NetworkFee);
+
+            var oracle = tx.GetAttribute<OracleResponse>();
+            if (oracle != null) oracleRequests.TryAdd(oracle.Id, tx.Hash);
         }
 
         public bool CheckTransaction(Transaction tx, StoreView snapshot)
         {
             BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
             BigInteger fee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
-            return balance >= fee;
+            if (balance < fee) return false;
+
+            var oracle = tx.GetAttribute<OracleResponse>();
+            if (oracle != null &&
+                (!oracleRequests.TryGetValue(oracle.Id, out var hash) || hash != tx.Hash))
+            {
+                return false;
```

Comment:
BTW, this can be replaced with generic "Conflicts" mechanism (#1991) that would also allow for proper transaction to have higher priority over fallback.